src/couchapps/ReqMgr/updates/updaterequest.js

```diff
@@ -78,9 +89,7 @@ function(doc, req) {
         doc[key] = newValues[key];
 
         // If key is RequestStatus, also update the transition
-        if (key == "RequestStatus") {
-            updateTransition();
-        }
+        updateTransition(key);
```

Comment:
Directly calling it is certainly cleaner, but there must be a performance penalty for executing L9-13 for every single key. How about we at least define keyAllowed and keyMap outside that function?

Reply:
I think L9-13 has to run every single key whether it is inside the function or not. The question is how much does it cost to call the function each time. It was true that the cost of calling function is adding additional baggage before. I think these days Javascript engine is much more efficient.  (Not sure with other language, though)
https://www.codereadability.com/performance-cost-javascript-function-call-and-foreach/

But if you think putting the condition check outside is cleaner, we can move that.